### ISSUE TITLE
Anyone being able to read out solar pump speed?

### ISSUE THREAD

**Issue by mdk2412** [openv/vcontrold#64]
At 6552 the status of the solar pump can be read (apparently only first byte). However I have not been able to get the actual speed (of the PWM control)  from this address (second byte). I tried using the following unit descriptions, but the result is always 0 %:

<unit name='Prozent Pumpe'>
<abbrev>PP</abbrev>
<calc get='(100*B1)/255' set='V'/>
<type>uchar</type>
<entity>%</entity>
</unit>
<unit name='Prozent zweites Byte ganzzahlig (Pumpe)'>
<abbrev>PR2</abbrev>
<calc get="B1" set="B1"/>
<type>uchar</type>
<entity>%</entity>
</unit>

In the main display of the device (343-F) the percentage of the solar pump is shown in the service menu.

Anyone having had success with the speed read-out? Maybe at another address?

**Comment by philverh** [comment]
As far as I know, addr 6552 returns an ON/OFF (1/0) value coded on one byte. 
I don't have the same device as you, and do not know where the pump speed may be stored. Is the display on the device really the pump speed, or the duty cycle (in percent) over some time period, like the last hour ?

**Comment by mdk2412** [comment]
Yes, the display on the device shows a very dynamic actual speed in %, reacting to the delta T of water storage and collector. It is consistent with the status value, i.e. 0 while the pump is off.

**Comment by h07d0q** [comment]
I'm also interested in the percentage of the solar pump PWM. Did you find a solution in the meantime?
